client/src/components/categorical/value.js

```diff
@@ -22,6 +21,32 @@ class CategoryValue extends React.Component {
     });
   };
 
+  shouldComponentUpdate = nextProps => {
+    const { props } = this;
+    const { metadataField, categoryIndex, categoricalSelection } = props;
+    const { categoricalSelection: newCategoricalSelection } = nextProps;
+
+    const diff = Object.keys(this.props).reduce((value, key) => {
```

Comment:
Why compute the difference of the entire props object, only to later check only for `world` and `colorAccessor` difference?   

Unless I am misreading the code (and apologies if I am), you could remove the computation of `diff` entirely (lines 29-35), and change the two line below (lines 45-46) to just read  `props.world !== nextProps.world || props.colorAccessor !== nextProps.colorAccessor`.

Seems simpler, clearer and less compute?

Reply:
Totally right, I originally computed the diff to see _what_ was different. Now that I know what I'm looking for I can just specifically look for that 👍